skcosmo/preprocessing/_flexible_scaler.py

```diff
@@ -0,0 +1,600 @@
+            self.sample_weight_ = _check_sample_weight(sample_weight, K, dtype=K.dtype)
+            self.sample_weight_ = self.sample_weight_ / np.sum(self.sample_weight_)
+        else:
+            self.sample_weight_ = sample_weight
+
+        if self.with_center:
+            if self.sample_weight_ is not None:
+                self.K_fit_rows_ = np.average(K, weights=self.sample_weight_, axis=0)
+                self.K_fit_all_ = np.average(
+                    self.K_fit_rows_, weights=self.sample_weight_
+                )
+            else:
+                super().fit(K, y)
+
+            K_pred_cols = np.average(Kc, weights=self.sample_weight_, axis=1)[
+                :, np.newaxis
+            ]
+        else:
+            self.K_fit_rows_ = np.zeros(Kc.shape[1])
+            self.K_fit_all_ = 0.0
+            K_pred_cols = np.zeros((Kc.shape[0], 1))
+
+        if self.with_trace:
+
+            Kc -= self.K_fit_rows_
+            Kc -= K_pred_cols
+            Kc += self.K_fit_all_
+
+            self.scale_ = np.trace(Kc) / Kc.shape[0]
+        else:
+            self.scale_ = 1.0
+
+        return self
+
+    def transform(self, K, copy=True):
+        """Center kernel matrix.
+
+        Parameters
+        ----------
+        K : ndarray of shape (n_samples1, n_samples2)
+            Kernel matrix.
+
+        copy : bool, default=True
+            Set to False to perform inplace computation.
+
+        Returns
+        -------
+        K_new : ndarray of shape (n_samples1, n_samples2)
+            Transformed array
+        """
+
+        check_is_fitted(self)
+        K = self._validate_data(K, copy=copy, dtype=FLOAT_DTYPES, reset=False)
+
+        if self.with_center:
+            K_pred_cols = np.average(K, weights=self.sample_weight_, axis=1)[
+                :, np.newaxis
+            ]
+        else:
+            K_pred_cols = np.zeros((K.shape[0], 1))
+
+        K -= self.K_fit_rows_
+        K -= K_pred_cols
+        K += self.K_fit_all_
+
+        return K / self.scale_
+
+    def fit_transform(self, K, y=None, sample_weight=None, copy=True, **fit_params):
+        r"""Fit to data, then transform it.
+
+        Parameters
+        ----------
+        K : ndarray of shape (n_samples, n_samples)
+            Kernel matrix.
+
+        y : None
+            Ignored.
+
+        sample_weight: ndarray of shape (n_samples,), default=None
+            Weights for each sample. Sample weighting can be used to center (and
+            scale) data using a weighted mean. Weights are internally normalized
+            before preprocessing.
+
+        \**fit_params:
+            necessary for compatibility with the functions of the
+            TransformerMixin class
+
+        Returns
+        -------
+        K_new : ndarray of shape (n_samples1, n_samples2)
+            Transformed array
+        """
+        self.fit(K, y, sample_weight=sample_weight)
+        return self.transform(K, copy)
```

Comment:
What will this accomplish that the super-class of `fit_transform` does not?

Reply:
There is no difference, except for `sample_weight` - here we have explicitly documented it and pass it as a parameter. In principle, we can delete this function, but I think it's nice to have a more specific implementation.

Reply:
Yes, since the sklearn `KernelCenterer` doesn't allow sample weights, we need a separate `fit_transform` here.